device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java

```diff
@@ -688,7 +694,7 @@ public void onMessageReceived(IotHubTransportMessage message)
     @Override
     public void onAuthenticationFailed(String deviceId, TransportException transportException)
     {
-        if (this.deviceClientConfigs.size() > 1)
```

Comment:
This line was the source of this bug. The assumption previously was that if there was only one device session on the connection, that the connection was not multiplexing. However, you can open a multiplexed connection with only one device session so this assumption was wrong

Reply:
For multiplexed connections, we don't drop the full connection just because a single device session drops, so it makes sense to do the same even when there is only one device registered